web/pgadmin/tools/debugger/static/js/components/DebuggerArgumentComponent.jsx

```diff
@@ -823,6 +823,11 @@ export default function DebuggerArgumentComponent({ debuggerInfo, restartDebug,
                   debuggerArgsSchema.current.sessData.aregsCollection.forEach((data) => {
 
                     if (skipStep) { return; }
+                    
+                    // Ensure that unchecked boolean checkboxes are set to false.
+                    if (data.type === 'boolean' && (data.value === undefined || data.value === '' || data.value === '0')) {
```

Comment:
This callback for onDataChange and you're changing data in this. This is anti-pattern.
The value probably should be changed after fetching from backend.